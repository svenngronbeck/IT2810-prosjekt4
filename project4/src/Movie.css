.movieContainerSmall{
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  min-width: 200px;
  width: 15%;
  margin: 0 2% 3% 2.4%;
  cursor: pointer;
  color: black;
  text-decoration: none;
}

.innerMovieContainerSmall{
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: center;
  padding: 7px;
  background-color: #f2f2f2;
  opacity: 0.8;
  border-radius: 4px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.8), 0 10px 30px 0 rgba(0, 0, 0, 0.8);
}

.movieContainerBig{
    width: 100%;
  margin: 0 0 1% 0;
  cursor: pointer;
  text-decoration: none;
  color: black;
}

.innerMovieContainerBig{
  display: flex;
  flex: 1;
  flex-direction: row;
  align-items: center;
  padding: 30px;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.8), 0 10px 30px 0 rgba(0, 0, 0, 0.8);
}

.movieImageSmall{
  height: 280px;
  width: 190px;
  margin-top: 2px;
  border-bottom: 1px solid #ccc;
  padding: 3px;
  background: rgba(224,224,224,0.2);
  border-radius: 3px;
}

.movieImageBig{
  height: 300px;
  min-width: 210px;
  padding: 20px;
  border-bottom-left-radius: 5px;
  border-top-left-radius: 5px;
  background-color: #f2f2f2;
  border-right: 1px solid #ccc;
}

.titleTextSmallShow{
  display: none;
}

.titleTextSmallHide{
  display: block;
  text-align: left;
  font-size: 17px;
  max-width: 190px  ;
  padding: 15px;
}

.movieInfoHide{
  display: none;
}

.movieInfoShow{
  display: flex;
  flex-direction: column;
  min-height: 300px;
  width: 100%;
  padding: 20px;
  background-color: #f2f2f2;
  border-bottom-right-radius: 5px;
  border-top-right-radius: 5px;
}

.bigMovieTitle{
  font-size: 25px;
  color: black;
  padding-bottom: 5px;
  border-bottom: 1px solid #ccc;
}

.movieDescription{
  margin-top: 10px;
}

.movieGenre{
  margin-top:auto;
  margin-bottom: 10px;

}

.movieRatings{
  display: flex;
  align-items: center;
  background-color: #e5e5e5;
  padding: 7px 0;
  font-weight: bold;
  border-radius: 4px;
}

.ratingImage{
  max-width: 50px;
  max-height: 30px;
  margin: 0 10px;
}

/*.imageAndInfo{
  display: flex;
  max-width: 100%;
  max-height: 85%;
}*/