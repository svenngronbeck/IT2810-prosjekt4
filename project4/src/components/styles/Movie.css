/*---Styles for gridview---*/

.movieContainerSmall{
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  min-width: 200px;
  width: 15%;
  margin: 0 2% 3% 2.4%;
}

.innerMovieContainerSmall{
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: center;
  padding: 7px;
  background-color: #f2f2f2;
  opacity: 0.8;
  border-radius: 4px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.8), 0 10px 30px 0 rgba(0, 0, 0, 0.8);
}

.movieImageSmall{
  height: 280px;
  width: 190px;
  margin-top: 2px;
  border-bottom: 1px solid #ccc;
  padding: 3px;
  background: rgba(224,224,224,0.2);
  border-radius: 3px;
}

.titleTextSmallHide{
  display: none;
}

.titleTextSmallShow{
  display: block;
  text-align: left;
  font-size: 17px;
  max-width: 190px;
  padding: 15px;
}

.smallImgImdbHide{
  display: none;
}

.smallImgImdbShow{
  margin-top:5px;
  display: flex;
  align-items: center;
}

/*---Styles for listview---*/

.movieContainerBig{
  width: 100%;
  margin: 0 0 1% 0;
}

.innerMovieContainerBig{
  display: flex;
  flex: 1;
  flex-direction: row;
  align-items: center;
  padding: 30px;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.8), 0 10px 30px 0 rgba(0, 0, 0, 0.8);
}

.expandMovie{
  align-self: flex-start;
  margin-top: -20px;
  margin-right: -20px;
  color: black;
  text-decoration: none;
}

.hideExpandMovie{
  display: none;
}

.expandImage{
  max-width: 20px;
  align-self: flex-start;

}

.hideExpandImage{
  display: none;
}

.movieImageBig{
  height: 300px;
  width: 210px;
  background-color: #f2f2f2;
  border-right: 1px solid #ccc;
  margin-right: 5px;
}

.movieLink{
  color: black;
  text-decoration: none;
  cursor: pointer;
}

.movieInfoHide{
  display: none;
}

.movieInfoShow{
  display: flex;
  flex-direction: column;
  min-height: 300px;
  width: 100%;
  padding: 20px;
  background-color: #f2f2f2;
  border-radius: 5px;
  cursor: default;
}

.bigMovieTitle{
  font-size: 25px;
  color: black;
  padding-bottom: 5px;
  border-bottom: 1px solid #ccc;
}

.movieExtraInfo{
  margin-top: auto;
  margin-bottom: 0;
}

.movieDescription{
  margin-top: 10px;
}

.movieActors, .runTime, h5{
  margin-top: 5px;
  margin-bottom: 0;
}
.movieGenre{
  margin-bottom: 10px;
}

.movieRatings{
  margin-top: auto;
  display: flex;
  align-items: center;
  background-color: #e5e5e5;
  font-weight: bold;
  border-radius: 5px;
  cursor: default;
}

.imdbRatings{
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.ratingImage{
  max-width: 50px;
  max-height: 30px;
}

.userRatings{
  display: flex;
  align-items: center;
  justify-content: space-around;
  background-color: #e5e5e5;
  font-size: 28px;
  margin-left: 0.5em;
  margin-bottom: 3px;
  border-right: 1px solid #ccc;
  border-left: 1px solid #ccc;
  padding: 0 10px;
}

.userRatingText{
  font-size: 15px;
  margin-right: 5px;
  margin-left: 5px;
}

.giveUserRating{
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 0 10px;
}