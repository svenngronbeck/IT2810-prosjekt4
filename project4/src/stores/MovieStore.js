import { observable, action } from "mobx";
import { Component } from 'react';
import axios from "axios";

//This is our store, where all the state is stored.
//Below are all the observable variables where the state is saved
class MovieStore extends Component {
    @observable fetchedMovies = 0;
    @observable expandMovie = false;
    @observable movies = [];
    @observable genre  = "All genres";
    @observable genres =
        ["All genres", "Western","Comedy", "Action", "Crime", "Drama", "Musical", "Thriller", "Animation", "Adventure", "Family",
            "Fantasy", "Documentary", "Sci-Fi", "Romance", "Biography", "Mystery", "Horror", "Music", "History",
            "War", "Short", "Film-Noir", "Sport", "Talk-Show", "News", "Reality-TV", "Game-Show"];
    @observable minRating = 1;
    @observable tempSearchString = "";
    @observable endpoint = "";
    @observable putEndpoint = "";
    @observable searchParam = "";
    @observable searchBarValue = "";
    @observable sortValue = "title";

    //-----FUNCTIONS-----//

    //Function to retrieve data from our api.
    //It fetches data together a given searchparam from the user.
    //Axios will try to retrieve data, and then store the data in the "movies"-list
    @action fetchMovieData = async () => {
        if (!this.searchParam){this.searchParam = "";}
        if (this.tempSearchString !== this.searchParam){
            this.fetchedMovies = 0;
            this.movies = [];
            this.tempSearchString = this.searchParam;
        }

        //If the genre is All genres, the genreparameter is not incluuded in the endpoint
        if(this.genre === "All genres"){
            this.endpoint = 'http://it2810-32.idi.ntnu.no:8080/movies/' + this.searchParam + "?startindex=" + this.fetchedMovies + '&threshold='+this.minRating + '&sort=' + this.sortValue;
        }
        else{
            this.endpoint = 'http://it2810-32.idi.ntnu.no:8080/movies/' + this.searchParam + "?startindex=" + this.fetchedMovies + '&genre='+this.genre + '&threshold='+this.minRating + '&sort=' + this.sortValue;
        }
        try{
            //Fetching from our database
            await axios.get(this.endpoint)
                .then(res => {
                    this.movies = this.movies.slice().concat(res.data);
                });
        } catch (error) {
            throw error;
        }
    };

    //Function to post data to the database (ratings)
    @action putMovieRating = async (movieID, rating) => {
        this.putEndpoint = 'http://it2810-32.idi.ntnu.no:8080/movies/'+movieID+'/'+rating;
        try{
            axios.put(this.putEndpoint);
        } catch (error) {
            throw error;
        }
    };

    //Function to set the searchParame when the user search for a movie
    //Fetches movies, movies are cleared so the user the the first results from the database
    @action setSearchParam = async (val) => {
        this.searchParam = val;
        this.fetchedMovies = 0;
        this.movies = [];
        await this.fetchMovieData();
    };

    @action setExpandMovie = (val) => {
        this.expandMovie = val;
    };

    @action setSearchBarValue = (val) => {
        this.searchBarValue = val;
    };

    //Function to set the genre when the user filters on a genre
    //Fetches movies, movies are cleared so the user the the first results from the database
    @action setGenre = async (val) => {
        this.genre = val;
        this.fetchedMovies = 0;
        this.movies = [];
        await this.fetchMovieData();
    };

    //Function to set the minrating when the sets the minrating for a movie
    //Fetches movies, movies are cleared so the user the the first results from the database
    @action setMinRating = async (val) => {
        this.minRating = val;
        this.fetchedMovies = 0;
        this.movies = [];
        await this.fetchMovieData();
    };

    //Function to clear all and fetch the first 20 movies from the database
    @action clearAll = async () => {
        this.minRating = 1;
        this.fetchedMovies = 0;
        this.movies = [];
        this.searchParam = "";
        this.sortValue = "title";
        this.genre = "All genres";
        if(document.getElementById("infoText")){
            document.getElementById("infoText").innerHTML = "Showing all movies";
        }
        await this.fetchMovieData();
    };

    //Function to increase number of fetched
    //Fetches movies, movies are cleared so the user the the first results from the database
    @action increaseFetchedMovies = async () =>{
        this.fetchedMovies += 20;
        await this.fetchMovieData();
    };

    //Function to set the sortvalue. (title or imdb)
    //Fetches movies, movies are cleared so the user the the first results from the database
    @action setSortValue = async (val) => {
        this.sortValue = val;
        this.fetchedMovies = 0;
        this.movies = [];
        await this.fetchMovieData();
    };
}

export default new MovieStore();