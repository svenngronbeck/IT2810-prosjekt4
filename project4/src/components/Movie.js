import React, { Component } from 'react';
import '../Movie.css';
import {inject, observer} from "mobx-react";
import ReplacementImage from "../assets/images/noimage.png";

@inject('movieStore')
@observer
class Movie extends Component {


    //Function to change the observable variable "expandMovie".
    //Changes between grid and listview
    handleExpandMovie(){
        this.props.changeView();
    }

  //-----RENDER----------
  render() {
    return (
        <div className={this.props.movieStore.expandMovie ? "movieContainerBig" : "movieContainerSmall"} >
            <div onClick={() => this.handleExpandMovie()} className={this.props.movieStore.expandMovie ? "innerMovieContainerBig" : "innerMovieContainerSmall"} id={this.props.id}>
                <a className="movieLink" href={"#"+this.props.id}>
                    <img alt="Img not found.." onError={(e) => {
                        e.target.src = ReplacementImage; //replacement image imported above
                    }} className={this.props.movieStore.expandMovie ? "movieImageBig" : "movieImageSmall"} src={this.props.poster}/>
                </a>
                <div className={this.props.movieStore.expandMovie ? "movieInfoShow" : "movieInfoHide"}>
                  <div className="bigMovieTitle">
                    {this.props.title}
                  </div>
                  <div className="movieDescription">
                    {this.props.plot}
                  </div>
                  <div className="movieGenre">
                    Genre: {this.props.genres}
                  </div>
                    {/*<div className="movieActors">
                    Actors: {this.props.actors}
                  </div>*/}
                  <div onClick={() => this.handleExpandMovie()} className="movieRatings">
                      <a href={"https://www.imdb.com/title/" + this.props.imdbId} target="_blank" rel="noopener noreferrer">
                          <img className="ratingImage" alt="" src={require("../assets/images/imdb.png")}/>
                      </a>
                      {" - " + this.props.imdbRating}
                  </div>
                </div>
                <a className="movieLink" href={"#"+this.props.id}>
                    <div className={this.props.movieStore.expandMovie ? "titleTextSmallHide" : "titleTextSmallShow"}>{this.props.title}</div>
                </a>
                <div className="rating">{this.props.imdb}</div>
            </div>
        </div>
    )
  }
}

export default Movie;
