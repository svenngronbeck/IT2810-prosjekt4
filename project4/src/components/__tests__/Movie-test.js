import React from 'react';
import renderer from 'react-test-renderer';
import Movie from "../Movie";
import MovieStore from "../../stores/MovieStore";

describe("MovieStore", () => {
    //Creating an instance of MovieStore
    const movieStore = MovieStore;




});