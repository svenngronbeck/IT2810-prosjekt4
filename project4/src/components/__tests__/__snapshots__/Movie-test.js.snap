// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieStore Movie renders correctly? 1`] = `
Array [
  <div
    className="react-notification-root"
  >
    <div
      className="notification-container-top-left"
    />
    <div
      className="notification-container-top-right"
    />
    <div
      className="notification-container-bottom-left"
    />
    <div
      className="notification-container-bottom-right"
    />
    <div
      className="notification-container-top-center"
    />
    <div
      className="notification-container-bottom-center"
    />
  </div>,
  <div
    className="movieContainerSmall"
  >
    <div
      className="innerMovieContainerSmall"
      onClick={[Function]}
    >
      <a
        className="movieLink"
        href="#undefined"
      >
        <img
          alt="Img not found.."
          className="movieImageSmall"
          onError={[Function]}
        />
      </a>
      <div
        className="movieInfoHide"
      >
        <div
          className="bigMovieTitle"
        />
        <div
          className="movieDescription"
        />
        <div
          className="movieGenre"
        >
          Genre: 
        </div>
        <div
          className="movieRatings"
          onClick={[Function]}
        >
          <div
            className="imdbRatings"
          >
            <a
              href="https://www.imdb.com/title/undefined"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt=""
                className="ratingImage"
                src="imdb.png"
              />
            </a>
            (undefined)
          </div>
          <div
            className="userRatings"
          >
            <p
              className="userRatingText"
            >
              User rating:
            </p>
            <div
              className="star-ratings"
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "position": "relative",
                }
              }
              title="0 Stars"
            >
              <svg
                className="star-grad"
                style={
                  Object {
                    "height": "0",
                    "position": "absolute",
                    "visibility": "hidden",
                    "width": "0",
                    "zIndex": "0",
                  }
                }
              >
                <defs>
                  <linearGradient
                    id="starGrad145431135508366"
                    x1="0%"
                    x2="100%"
                    y1="0%"
                    y2="0%"
                  >
                    <stop
                      className="stop-color-first"
                      offset="0%"
                      style={
                        Object {
                          "stopColor": "#eac600",
                          "stopOpacity": "1",
                        }
                      }
                    />
                    <stop
                      className="stop-color-first"
                      offset="0%"
                      style={
                        Object {
                          "stopColor": "#eac600",
                          "stopOpacity": "1",
                        }
                      }
                    />
                    <stop
                      className="stop-color-final"
                      offset="0%"
                      style={
                        Object {
                          "stopColor": "rgb(203, 211, 227)",
                          "stopOpacity": "1",
                        }
                      }
                    />
                    <stop
                      className="stop-color-final"
                      offset="100%"
                      style={
                        Object {
                          "stopColor": "rgb(203, 211, 227)",
                          "stopOpacity": "1",
                        }
                      }
                    />
                  </linearGradient>
                </defs>
              </svg>
              <div
                className="star-container"
                onClick={null}
                onMouseEnter={null}
                onMouseLeave={null}
                style={
                  Object {
                    "cursor": undefined,
                    "display": "inline-block",
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "position": "relative",
                    "verticalAlign": "middle",
                  }
                }
              >
                <svg
                  className="widget-svg"
                  style={
                    Object {
                      "height": "20px",
                      "transform": undefined,
                      "transition": "transform .2s ease-in-out",
                      "width": "20px",
                    }
                  }
                  viewBox="0 0 51 48"
                >
                  <path
                    className="star"
                    d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                    style={
                      Object {
                        "fill": "rgb(203, 211, 227)",
                        "transition": "fill .2s ease-in-out",
                      }
                    }
                  />
                </svg>
              </div>
            </div>
            <p
              className="userRatingText"
            >
              (
              NaN
              ) - 
               votes
            </p>
          </div>
          <div
            className="giveUserRating"
          >
            <p
              className="userRatingText"
            >
              Give user-rating: 
            </p>
            <div
              className="star-ratings"
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "position": "relative",
                }
              }
              title="0 Stars"
            >
              <svg
                className="star-grad"
                style={
                  Object {
                    "height": "0",
                    "position": "absolute",
                    "visibility": "hidden",
                    "width": "0",
                    "zIndex": "0",
                  }
                }
              >
                <defs>
                  <linearGradient
                    id="starGrad264697042354166"
                    x1="0%"
                    x2="100%"
                    y1="0%"
                    y2="0%"
                  >
                    <stop
                      className="stop-color-first"
                      offset="0%"
                      style={
                        Object {
                          "stopColor": "#eac600",
                          "stopOpacity": "1",
                        }
                      }
                    />
                    <stop
                      className="stop-color-first"
                      offset="0%"
                      style={
                        Object {
                          "stopColor": "#eac600",
                          "stopOpacity": "1",
                        }
                      }
                    />
                    <stop
                      className="stop-color-final"
                      offset="0%"
                      style={
                        Object {
                          "stopColor": "rgb(203, 211, 227)",
                          "stopOpacity": "1",
                        }
                      }
                    />
                    <stop
                      className="stop-color-final"
                      offset="100%"
                      style={
                        Object {
                          "stopColor": "rgb(203, 211, 227)",
                          "stopOpacity": "1",
                        }
                      }
                    />
                  </linearGradient>
                </defs>
              </svg>
              <div
                className="star-container"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "display": "inline-block",
                    "paddingLeft": undefined,
                    "paddingRight": "2px",
                    "position": "relative",
                    "verticalAlign": "middle",
                  }
                }
              >
                <svg
                  className="widget-svg"
                  style={
                    Object {
                      "height": "20px",
                      "transform": undefined,
                      "transition": "transform .2s ease-in-out",
                      "width": "20px",
                    }
                  }
                  viewBox="0 0 51 48"
                >
                  <path
                    className="star"
                    d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                    style={
                      Object {
                        "fill": "rgb(203, 211, 227)",
                        "transition": "fill .2s ease-in-out",
                      }
                    }
                  />
                </svg>
              </div>
              <div
                className="star-container"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "display": "inline-block",
                    "paddingLeft": "2px",
                    "paddingRight": "2px",
                    "position": "relative",
                    "verticalAlign": "middle",
                  }
                }
              >
                <svg
                  className="widget-svg"
                  style={
                    Object {
                      "height": "20px",
                      "transform": undefined,
                      "transition": "transform .2s ease-in-out",
                      "width": "20px",
                    }
                  }
                  viewBox="0 0 51 48"
                >
                  <path
                    className="star"
                    d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                    style={
                      Object {
                        "fill": "rgb(203, 211, 227)",
                        "transition": "fill .2s ease-in-out",
                      }
                    }
                  />
                </svg>
              </div>
              <div
                className="star-container"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "display": "inline-block",
                    "paddingLeft": "2px",
                    "paddingRight": "2px",
                    "position": "relative",
                    "verticalAlign": "middle",
                  }
                }
              >
                <svg
                  className="widget-svg"
                  style={
                    Object {
                      "height": "20px",
                      "transform": undefined,
                      "transition": "transform .2s ease-in-out",
                      "width": "20px",
                    }
                  }
                  viewBox="0 0 51 48"
                >
                  <path
                    className="star"
                    d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                    style={
                      Object {
                        "fill": "rgb(203, 211, 227)",
                        "transition": "fill .2s ease-in-out",
                      }
                    }
                  />
                </svg>
              </div>
              <div
                className="star-container"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "display": "inline-block",
                    "paddingLeft": "2px",
                    "paddingRight": "2px",
                    "position": "relative",
                    "verticalAlign": "middle",
                  }
                }
              >
                <svg
                  className="widget-svg"
                  style={
                    Object {
                      "height": "20px",
                      "transform": undefined,
                      "transition": "transform .2s ease-in-out",
                      "width": "20px",
                    }
                  }
                  viewBox="0 0 51 48"
                >
                  <path
                    className="star"
                    d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                    style={
                      Object {
                        "fill": "rgb(203, 211, 227)",
                        "transition": "fill .2s ease-in-out",
                      }
                    }
                  />
                </svg>
              </div>
              <div
                className="star-container"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "display": "inline-block",
                    "paddingLeft": "2px",
                    "paddingRight": undefined,
                    "position": "relative",
                    "verticalAlign": "middle",
                  }
                }
              >
                <svg
                  className="widget-svg"
                  style={
                    Object {
                      "height": "20px",
                      "transform": undefined,
                      "transition": "transform .2s ease-in-out",
                      "width": "20px",
                    }
                  }
                  viewBox="0 0 51 48"
                >
                  <path
                    className="star"
                    d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                    style={
                      Object {
                        "fill": "rgb(203, 211, 227)",
                        "transition": "fill .2s ease-in-out",
                      }
                    }
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="smallImgStarShow"
      >
        <div
          className="star-ratings"
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "position": "relative",
            }
          }
          title="0 Stars"
        >
          <svg
            className="star-grad"
            style={
              Object {
                "height": "0",
                "position": "absolute",
                "visibility": "hidden",
                "width": "0",
                "zIndex": "0",
              }
            }
          >
            <defs>
              <linearGradient
                id="starGrad860924933330522"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              >
                <stop
                  className="stop-color-first"
                  offset="0%"
                  style={
                    Object {
                      "stopColor": "#eac600",
                      "stopOpacity": "1",
                    }
                  }
                />
                <stop
                  className="stop-color-first"
                  offset="0%"
                  style={
                    Object {
                      "stopColor": "#eac600",
                      "stopOpacity": "1",
                    }
                  }
                />
                <stop
                  className="stop-color-final"
                  offset="0%"
                  style={
                    Object {
                      "stopColor": "rgb(203, 211, 227)",
                      "stopOpacity": "1",
                    }
                  }
                />
                <stop
                  className="stop-color-final"
                  offset="100%"
                  style={
                    Object {
                      "stopColor": "rgb(203, 211, 227)",
                      "stopOpacity": "1",
                    }
                  }
                />
              </linearGradient>
            </defs>
          </svg>
          <div
            className="star-container"
            onClick={null}
            onMouseEnter={null}
            onMouseLeave={null}
            style={
              Object {
                "cursor": undefined,
                "display": "inline-block",
                "paddingLeft": undefined,
                "paddingRight": "2px",
                "position": "relative",
                "verticalAlign": "middle",
              }
            }
          >
            <svg
              className="widget-svg"
              style={
                Object {
                  "height": "15px",
                  "transform": undefined,
                  "transition": "transform .2s ease-in-out",
                  "width": "15px",
                }
              }
              viewBox="0 0 51 48"
            >
              <path
                className="star"
                d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                style={
                  Object {
                    "fill": "rgb(203, 211, 227)",
                    "transition": "fill .2s ease-in-out",
                  }
                }
              />
            </svg>
          </div>
          <div
            className="star-container"
            onClick={null}
            onMouseEnter={null}
            onMouseLeave={null}
            style={
              Object {
                "cursor": undefined,
                "display": "inline-block",
                "paddingLeft": "2px",
                "paddingRight": "2px",
                "position": "relative",
                "verticalAlign": "middle",
              }
            }
          >
            <svg
              className="widget-svg"
              style={
                Object {
                  "height": "15px",
                  "transform": undefined,
                  "transition": "transform .2s ease-in-out",
                  "width": "15px",
                }
              }
              viewBox="0 0 51 48"
            >
              <path
                className="star"
                d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                style={
                  Object {
                    "fill": "rgb(203, 211, 227)",
                    "transition": "fill .2s ease-in-out",
                  }
                }
              />
            </svg>
          </div>
          <div
            className="star-container"
            onClick={null}
            onMouseEnter={null}
            onMouseLeave={null}
            style={
              Object {
                "cursor": undefined,
                "display": "inline-block",
                "paddingLeft": "2px",
                "paddingRight": "2px",
                "position": "relative",
                "verticalAlign": "middle",
              }
            }
          >
            <svg
              className="widget-svg"
              style={
                Object {
                  "height": "15px",
                  "transform": undefined,
                  "transition": "transform .2s ease-in-out",
                  "width": "15px",
                }
              }
              viewBox="0 0 51 48"
            >
              <path
                className="star"
                d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                style={
                  Object {
                    "fill": "rgb(203, 211, 227)",
                    "transition": "fill .2s ease-in-out",
                  }
                }
              />
            </svg>
          </div>
          <div
            className="star-container"
            onClick={null}
            onMouseEnter={null}
            onMouseLeave={null}
            style={
              Object {
                "cursor": undefined,
                "display": "inline-block",
                "paddingLeft": "2px",
                "paddingRight": "2px",
                "position": "relative",
                "verticalAlign": "middle",
              }
            }
          >
            <svg
              className="widget-svg"
              style={
                Object {
                  "height": "15px",
                  "transform": undefined,
                  "transition": "transform .2s ease-in-out",
                  "width": "15px",
                }
              }
              viewBox="0 0 51 48"
            >
              <path
                className="star"
                d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                style={
                  Object {
                    "fill": "rgb(203, 211, 227)",
                    "transition": "fill .2s ease-in-out",
                  }
                }
              />
            </svg>
          </div>
          <div
            className="star-container"
            onClick={null}
            onMouseEnter={null}
            onMouseLeave={null}
            style={
              Object {
                "cursor": undefined,
                "display": "inline-block",
                "paddingLeft": "2px",
                "paddingRight": undefined,
                "position": "relative",
                "verticalAlign": "middle",
              }
            }
          >
            <svg
              className="widget-svg"
              style={
                Object {
                  "height": "15px",
                  "transform": undefined,
                  "transition": "transform .2s ease-in-out",
                  "width": "15px",
                }
              }
              viewBox="0 0 51 48"
            >
              <path
                className="star"
                d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                style={
                  Object {
                    "fill": "rgb(203, 211, 227)",
                    "transition": "fill .2s ease-in-out",
                  }
                }
              />
            </svg>
          </div>
        </div>
      </div>
      <a
        className="movieLink"
        href="#undefined"
      >
        <div
          className="titleTextSmallShow"
        />
      </a>
      <div
        className="rating"
      />
      <img
        alt=""
        className="hideExpandImage"
        src="expanded.png"
      />
    </div>
  </div>,
]
`;
